lib/ansible/modules/monitoring/uptimerobot.py

```diff
@@ -27,124 +20,168 @@
-
-    params['monitorStatus'] = 0
-    data = urlencode(params)
-    full_uri = API_BASE + API_ACTIONS['editMonitor'] + data
-    req, info = fetch_url(module, full_uri)
-    result = req.read()
-    jsonresult = json.loads(result)
-    req.close()
-    return jsonresult['stat']
+class UptimeRobot:
+    def __init__(self, params, api_methods):
+        self.headers = {}
+        self.headers['Content-Type'] = "application/json"
+        self.uri = ''
+        self.api_methods = api_methods
+        self.params = params
+        self.params = self.check_dict(self.params)
+        for k, v in self.params.items():
+            if k == 'type':
+                if v == 'dns':
+                    self.params[k] = 2
+                else:
+                    self.params[k] = 1
+        self.body = json.dumps(self.params)
+        for state, method in self.api_methods.items():
+            if state == self.params['status']:
+                self.api_method = method
+
+    def get_monitor_id(self, fname):
+        monitors = self.get_monitors()
+        for m_dict in monitors['monitors']:
+            if m_dict['friendly_name'] == fname:
+                mid = m_dict['id']
+                return mid
+
+    def get_monitors(self):
+        data = {}
+        data['api_key'] = self.params['api_key']
+        data = json.dumps(data)
+        self.uri = API_BASE + "getMonitors"
+        req = Request()
+        state = req.post(url=self.uri, data=data, headers=self.headers)
+        state = state.read()
+        state = json.loads(state)
+        return state
+
+    def api_request(self):
+        self.uri = API_BASE + self.api_method
+        req = Request()
+        state = req.post(url=self.uri, data=self.body, headers=self.headers)
+        state = state.read()
+        state = json.loads(state)
+        return state, self.api_method
+
+    def check_dict(self, dictionary):
+        tmp = {}
+        for k, v in dictionary.items():
+            if v != '':
+                tmp[k] = v
+        if dictionary['status'] != 'created':
+            tmp['id'] = self.get_monitor_id(dictionary['friendly_name'])
+        dictionary.clear()
+        return tmp
 
 
 def main():
+    result = dict(
+        changed=False,
+        original_message='',
+        message=''
+    )
+
+    api_methods = dict(
+        started='editMonitor',
+        paused='editMonitor',
+        absent='deleteMonitor',
+        created='newMonitor'
+    )
 
     module = AnsibleModule(
         argument_spec=dict(
-            state=dict(required=True, choices=['started', 'paused']),
+            name=dict(required=True),
+            url=dict(required=True),
+            check_type=dict(required=False, choices=['http', 'dns']),
+            state=dict(required=True, choices=['started', 'paused', 'absent', 'created']),
             apikey=dict(required=True, no_log=True),
-            monitorid=dict(required=True)
+            monitorid=dict(required=False)
```

Comment:
Add a note in documentation stating that `monitorid` is not required parameter from 2.10 and a porting guide note [here](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/porting_guides/porting_guide_2.10.rst#noteworthy-module-changes)

Reply:
https://github.com/Ninjiner/ansible/commit/f2c2822c08a7bac19e0a0d111e63cc709e801b97

Reply:
https://github.com/Ninjiner/ansible/commit/36e2719a1a82e24c04c34dc03e4115c2c8903da9